File: SortFraction.py
from AddFraction import PlusFraction

# A class with a parent
class ArrangeFraction(PlusFraction):

    # A constructor
    def __init__(self, fractions):
        super().__init__(fractions)

        self.final_numerators   = []
        self.final_denominators = []


    # Returns a dictionary of the new fraction
    def sortAscending(self):
		# STEPS 1, 2:
        self.canonizeFraction()

        self.copy_numerators = []
        for member in self.new_numerators: self.copy_numerators.append(member);

		# STEP 3:
        # the little difference lies here(sort ascending)
        self.copy_numerators.sort()

        # map sorted (transformed) fractions to the original ones
        for numerator in self.copy_numerators:
            # get index using list value
            self.position = self.new_numerators.index(numerator)
            self.final_numerators.append(self.numerators[self.position])
            self.final_denominators.append(self.denominators[self.position])
                    
        return {'numerators':self.final_numerators, 'denominators':self.final_denominators}

    # Returns a dictionary of the new fraction
    def sortDescending(self):
		# STEPS 1, 2:
        self.canonizeFraction()

        self.copy_numerators = []
        for member in self.new_numerators: self.copy_numerators.append(member);

		# STEP 3:
        # the little difference lies here(sort ascending)
        self.copy_numerators.sort(reverse=True)

        # map sorted (transformed) fractions to the original ones
        for numerator in self.copy_numerators:
            # get index using list value
            self.position = self.new_numerators.index(numerator)
            self.final_numerators.append(self.numerators[self.position])
            self.final_denominators.append(self.denominators[self.position])
                    
        return {'numerators':self.final_numerators, 'denominators':self.final_denominators}

File: SortFraction.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Algebra_CS
{
    class SortFraction : AddFraction
    {
        public List<int> final_numerators;
        public List<int> final_denominators;

        public SortFraction(List<int> num, List<int> denom) : base(num, denom)
        {
            final_numerators = new List<int>();
            final_denominators = new List<int>();
        }

        public void sortAscending()
        {
            List<int> copy_numerators = new List<int>();
            int index;

            // STEPS 1, 2:
            canonizeFraction();

            foreach (int nn in new_numerators)
            {
                copy_numerators.Add(nn);
            }

            // STEP 3:
            // Sort array in ascending order order
            copy_numerators.Sort();

            // map sorted transformed fractions to their originals
            foreach (int sorted in copy_numerators)
            {
                index = Array.IndexOf<int>(new_numerators.ToArray<int>(), sorted);
                final_numerators.Add(numerators[index]);
                final_denominators.Add(denominators[index]);
            }
        }

        public void sortDescending()
        {
            List<int> copy_numerators = new List<int>();
            int index;

            // STEPS 1, 2:
            canonizeFraction();

            foreach (int nn in new_numerators)
            {
                copy_numerators.Add(nn);
            }

            // STEP 3:
            // Sort array in descending order order
            copy_numerators.Sort();
            copy_numerators.Reverse();

            // map sorted transformed fractions to their originals
            foreach (int sorted in copy_numerators)
            {
                index = Array.IndexOf<int>(new_numerators.ToArray<int>(), sorted);
                final_numerators.Add(numerators[index]);
                final_denominators.Add(denominators[index]);
            }
        }
    }
}
